tests/src/Kernel/Access/AccessByOgMembershipTest.php

```diff
@@ -129,15 +130,16 @@ public function setUp() {
 
     // Grant both members and non-members permission to edit any group content.
     foreach ([OgRoleInterface::AUTHENTICATED, OgRoleInterface::ANONYMOUS] as $role_name) {
-      $role_id = "block_content-group-$role_name";
-      /** @var \Drupal\og\Entity\OgRole $role */
-      $role = $this->container->get('entity_type.manager')
-        ->getStorage('og_role')
-        ->load($role_id);
-      $role->grantPermission('edit any group_content content');
-      $role->save();
+      /** @var OgRole $role */
+      $role = OgRole::getRole('block_content', 'group', $role_name);
```

Comment:
@dakala I've re-added saving the roles for non-members as-well (and even used the new `OgRole::getRole`). The error was that a few lines below we didn't re-load the $role for the auth user, thus saved the membership with the wrong role.